lib/cryptoexchange/exchanges/kraken_future/services/contract_stat.rb

```diff
@@ -9,26 +9,72 @@ def supports_individual_ticker_query?
           end
         end
 
+      end_time = 'https://www.cryptofacilities.com/derivatives/api/v3/instruments'
+      funding_rate = 'https://www.cryptofacilities.com/derivatives/api/v3/tickers'
+
         def fetch(market_pair)
           output = super(open_interest_url)
-          adapt(output, market_pair)
+          contract_info = get_contract_info(market_pair)
+          adapt(output, contract_info, market_pair)
+        end
+
+        def get_contract_info(market_pair)
+          url = if market_pair.contract_interval == "perpetual"
+                  "#{Cryptoexchange::Exchanges::KrakenFutures::Market::API_URL}/tickers"
+                else
+                  "#{Cryptoexchange::Exchanges::KrakenFutures::Market::API_URL}/instruments"
+                end
+          contract_info = HTTP.get(url).parse(:json)
+          process_contract_info(contract_info, market_pair)
+        end
+
+        def process_contract_info(contract_info, market_pair)
+          arr = {}
+          if market_pair.contract_interval == "perpetual"
+            contract = contract_info['tickers'].find { |i| i['symbol'] == market_pair.inst_id.downcase }
+            arr["funding_rate_percentage"] = contract["fundingRate"].to_f * 100
+            arr["funding_rate_percentage_predicted"] = contract["fundingRatePrediction"].to_f * 100
+            arr["next_funding_timestamp"] = nil
+          else
+            contract = contract_info['instruments'].find { |i| i["symbol"] == market_pair.inst_id.downcase }
+            arr["expire_timestamp"] = DateTime.parse(contract["lastTradingTime"]).to_time.to_i
```

Comment:
curious do we still need `to_time` ?

Reply:
yea, you need parse the `to_time` first before you can convert the time into timestamp or you will get `NoMethodError`